src/components/molecules/PageNation/index.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <ul class="page-nation-lists">
+    <template v-if="4 <= currentPage">
+      <li class="page-nation-list">
+        <app-button @click="$emit('handle-page-button-click', 1)">
+          1
+        </app-button>
+      </li>
+      <li class="page-nation-list page-nation-list-dots">…</li>
+    </template>
+    <li
+      v-for="page in pageRange"
+      :key="page"
+      class="page-nation-list"
+    >
+      <app-button
+        :disabled="currentPage === page"
+        @click="$emit('handle-page-button-click', page)"
+      >
+        {{ page }}
+      </app-button>
+    </li>
+    <template v-if="currentPage <= totalPages - 3">
+      <li class="page-nation-list page-nation-list-dots">…</li>
+      <li class="page-nation-list">
+        <app-button @click="$emit('handle-page-button-click', totalPages)">
+          {{ totalPages }}
+        </app-button>
+      </li>
+    </template>
+  </ul>
+</template>
+
+<script>
+import { Button } from '@Components/atoms';
+
+export default {
+  components: {
+    appButton: Button,
+  },
+  computed: {
+    currentPage() {
+      return this.$store.state.articles.pageData.currentPage;
+    },
+    totalPages() {
+      return this.$store.state.articles.pageData.totalPages;
+    },
+    pageRange() {
+      if (this.totalPages === null) {
+        return false;
+      }
+      const range = [...Array(this.totalPages).keys()].map(i => i + 1);
+      if (this.currentPage <= 3) {
+        const result = range.slice(0, 5);
+        return result;
+      }
+      if (this.totalPages - 2 <= this.currentPage) {
+        const result = range.slice(this.totalPages - 5);
```

Comment:
[nits]
後ろから5つ取得したい時は
```suggestion
        const result = range.slice(-5);
```
でも問題ないです！

Reply:
処理を簡略化しました。
[36400e2](https://github.com/gizumo-education/gizumo_wiki_lesson/pull/73/commits/36400e2f258cd5ba373d0930b41cd3bfd9afa56a)